src/test/java/org/jenkinsci/plugins/gitclient/CliGitAPITempFileTest.java

```diff
@@ -0,0 +1,123 @@
+package org.jenkinsci.plugins.gitclient;
+
+import hudson.EnvVars;
+import hudson.model.TaskListener;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.jvnet.hudson.test.Issue;
+
+/**
+ * Test that createTempFile is adapting its directory name choices to match
+ * platform limitations of command line git.
+ *
+ * @author Mark Waite
+ */
+@RunWith(Parameterized.class)
+public class CliGitAPITempFileTest {
+
+    private final String workspaceDirName;
+    private final boolean mustUseSystemTempDir;
+    /* Should temp folder be in same parent dir as workspace? */
+
+    private static final String INVALID_CHARACTERS = "%" + (isWindows() ? " ()" : "`");
+
+    @Rule
+    public TemporaryFolder workspaceParentFolder = new TemporaryFolder();
+
+    public CliGitAPITempFileTest(String workspaceDirName, boolean mustUseSystemTempDir) {
+        this.workspaceDirName = workspaceDirName;
+        this.mustUseSystemTempDir = mustUseSystemTempDir;
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection workspaceDirNames() {
```

Comment:
That is a parameterized JUnit test.  The Parameterized annotation causes the JUnit executor to call this method to assemble the arguments to the constructor for this test class.

It allows me to readily construct many different cases which all use the same assertion.  This particular case runs about 10 tests, one for each of the different workspace directory names in the workspaceDirNames method